src/pykeen/sampling/negative_sampler.py

```diff
@@ -21,7 +21,9 @@ class NegativeSampler(ABC):
     """A negative sampler."""
 
     #: The default strategy for optimizing the negative sampler's hyper-parameters
-    hpo_default: ClassVar[Mapping[str, Mapping[str, Any]]]
+    hpo_default: ClassVar[Mapping[str, Mapping[str, Any]]] = dict(
+        num_negs_per_pos=dict(type=int, low=0, high=2, scale='power', base=10),
```

Comment:
This means we only have three different `num_neg_per_pos` values by default, right?

Reply:
https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial.suggest_int

with `log=True` seems to be more appropriate to me. It still explores different scales uniformly, but does not "lose" possible `num_neg_per_pos` values

Reply:
True, but are there real meaningful differences in exploring the difference between 2 and 3, or 50 and 60? I like the current solution as a way of significantly reducing the possible space to explore.

We could go either way, though. I guess we should bake in the ability to use log anyway

Reply:
Updated in 2440c5b2